[BUG] test_yyyyMMdd_format_for_legacy_mode[DATAGEN_SEED=1727619674, TZ=UTC] failed GPU and CPU are not both null

**Describe the bug**
first seen in rapids_integration-scala213-pre_release-github, run: 127 but no other occurrence.

It could be a failure when specific DATAGEN_SEED=1727619674.
```
[2024-09-29T16:31:20.704Z] FAILED ../../src/main/python/date_time_test.py::test_yyyyMMdd_format_for_legacy_mode[DATAGEN_SEED=1727619674, TZ=UTC] - AssertionError: GPU and CPU are not both null at [1733, 'unix_timestamp(a, yyyyMMdd)']
[2024-09-29T16:31:20.705Z] = 1 failed, 30669 passed, 1886 skipped, 598 xfailed, 658 xpassed, 15102 warnings in 7805.95s (2:10:05) =
```

```
[2024-09-29T16:31:20.703Z]         elif (cpu == None):
[2024-09-29T16:31:20.703Z] >           assert cpu == gpu, "GPU and CPU are not both null at {}".format(path)
[2024-09-29T16:31:20.703Z] E           AssertionError: GPU and CPU are not both null at [1733, 'unix_timestamp(a, yyyyMMdd)']
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/date_time_test.py
+++ b/integration_tests/src/main/python/date_time_test.py
@@ -460,8 +460,9 @@ def test_to_timestamp(parser_policy):
         { "spark.sql.legacy.timeParserPolicy": parser_policy})
 
 @pytest.mark.skipif(not is_supported_time_zone(), reason="not all time zones are supported now, refer to https://github.com/NVIDIA/spark-rapids/issues/6839, please update after all time zones are supported")
+# Test years after 1900, refer to issues: https://github.com/NVIDIA/spark-rapids/issues/11543, https://github.com/NVIDIA/spark-rapids/issues/11539
 def test_yyyyMMdd_format_for_legacy_mode():
-    gen = StringGen("[0-9]{3}[1-9](0[1-9]|1[0-2])(0[1-9]|[1-2][0-9])")
+    gen = StringGen('(19[0-9]{2}|[2-9][0-9]{3})([0-9]{4})')
     assert_gpu_and_cpu_are_equal_sql(
         lambda spark : unary_op_df(spark, gen),
         "tab",
@@ -470,8 +471,8 @@ def test_yyyyMMdd_format_for_legacy_mode():
                   date_format(to_timestamp(a, 'yyyyMMdd'), 'yyyyMMdd')
            from tab
         ''',
-        {  'spark.sql.legacy.timeParserPolicy': 'LEGACY',
-           'spark.rapids.sql.incompatibleDateFormats.enabled': True})
+        {'spark.sql.legacy.timeParserPolicy': 'LEGACY',
+         'spark.rapids.sql.incompatibleDateFormats.enabled': True})
 
 @tz_sensitive_test
 @pytest.mark.skipif(not is_supported_time_zone(), reason="not all time zones are supported now, refer to https://github.com/NVIDIA/spark-rapids/issues/6839, please update after all time zones are supported")

